physicsnemo/models/diffusion/sfm_preconditioning.py

```diff
@@ -0,0 +1,239 @@
+            Maximum noise level
+        """
+        ema_weight = self.ema_weight.to(self.sigma_max_current.device)
+        sigma_max = torch.tensor(sigma_max).to(self.sigma_max_current.device)
+        # Update sigma_max_current without gradients
+        new_sigma_max_current = (
+            ema_weight * self.sigma_max_current + (1 - ema_weight) * sigma_max
+        )
+        self.sigma_max_current = torch.max(
+            new_sigma_max_current, self.min_values.to(self.sigma_max_current.device)
+        )
+
+    def get_sigma_max(self):
+        """returns the current max sigma"""
+        return self.sigma_max_current
+
+    @nvtx.annotate(message="SFMPrecondSR", color="orange")
+    def forward(
+        self,
+        x: torch.Tensor,
+        sigma: torch.Tensor,
+        condition: torch.Tensor,
+        force_fp32: bool = False,
+        **model_kwargs,
+    ):
+        """
+        Forward pass of the Stochastic Flow Model preconditioner
+
+        Parameters
+        ----------
+        x : tensor
+            The partially noised input image
+
+        sigma : torch.Tensor
+            The image containing random noise
+
+        condition : torch.Tensor
+            The low resoltuion
+
+        force_fp32 : bool
+            Whether float 32 computations should be forced, default False
+
+        model_kwargs: dict
+            Keyword arguments for the underlying model.
+
+        Returns
+        -------
+        torch.Tensor : the denoised image
+
+        """
+        x = x.to(torch.float32)
+        dtype = (
+            torch.float16
+            if (self.use_fp16 and not force_fp32 and x.device.type == "cuda")
+            else torch.float32
+        )
+        c_skip = self.sigma_data**2 / (sigma**2 + self.sigma_data**2)
+        c_out = sigma * self.sigma_data / (sigma**2 + self.sigma_data**2).sqrt()
+        c_in = 1 / (self.sigma_data**2 + sigma**2).sqrt()
+        c_noise = sigma.log() / 4
+
+        if not self.use_x_low_conditioning:
+            scaled_x = c_in * x
+        else:
+            condition = condition.to(torch.float32)
+            scaled_x = torch.cat([c_in * x, condition], dim=1)
+
+        F_x = self.denoiser_net(
+            scaled_x.to(dtype), noise_labels=c_noise.flatten(), class_labels=None
+        )
+
+        D_x = c_skip * x + c_out * F_x.to(torch.float32)
+        return D_x
+
+    def round_sigma(self, sigma: Union[List[float], torch.Tensor]):
+        """
+        Convert a given sigma value(s) to a tensor representation in the
+                same precision as the model
+
+        Parameters
+        ----------
+        sigma : Union[float list, torch.Tensor]
+            The sigma value(s) to convert.
+
+        Returns
+        -------
+        torch.Tensor
+            The tensor representation of the provided sigma value(s).
+        """
+        return torch.as_tensor(sigma)
+
+
+class SFMPrecondEmpty(Module):
+    """
+    A preconditioner that does nothing
+
```

Comment:
For `SFMPrecondEmpty` I suggest just adding a simpler one-liner `nn.Module` or something in the train script and avoid putting this in the core `physicsnemo` package code. Since the need for it is really example-specific and only supports the edge case of training an encoder-only model in that example (I assume for a point of comparison against the real SFM model which has encoder/denoiser trained jointly)